presto-main-base/src/main/java/com/facebook/presto/metadata/FunctionSignatureMatcher.java

```diff
@@ -71,7 +71,7 @@ public Optional<Signature> match(Collection<? extends SqlFunction> candidates, L
                 .filter(function -> !function.getSignature().getTypeVariableConstraints().isEmpty())
                 .collect(Collectors.toList());
 
-        match = matchFunctionExact(genericCandidates, parameterTypes);
+        match = matchFunctionGeneric(genericCandidates, parameterTypes);
```

Comment:
Separate this into a PR also for just this condition and write an individual test for it.